src/core/qgsvectorfilewriter.cpp

```diff
@@ -237,6 +237,21 @@ void QgsVectorFileWriter::init( QString vectorFileName,
     mOgrDriverName = driverName;
   }
 
+  QString fidFieldName;
```

Comment:
could be protected under #if GDAL_VERSION_NUM < GDAL_COMPUTE_VERSION(3,3,1) as well. This will make future clean-up easier.
(and the "defined(GDAL_COMPUTE_VERSION) &&" test is not needed given our base requirement for GDAL)